lib/seahorn/HornCex.cc

```diff
@@ -220,11 +227,25 @@ namespace seahorn
               << ". Expected .xml, .ll, or .bc.\n";
     }
 
-    if (!SliceOutputFile.empty()) {
-      sliceErrorTrace(F, trace);
-      dumpLLVMBitcode(M, SliceOutputFile.c_str());
+    if (!BmcSliceOutputFile.empty()) {
+      llvm::DenseSet<const llvm::BasicBlock*> region(trace.getRegion());
+      reduceToRegion(F, region, false);
+      dumpLLVMBitcode(M, BmcSliceOutputFile.c_str());
       return true;
     }
+
+    if (!CpSliceOutputFile.empty())
+    {
+      DenseSet<const BasicBlock*> region;
+      for (const CutPoint* cp:cpTrace)
+      for (CutPoint::const_iterator it=cp->succ_begin();it!=cp->succ_end();it++)
+      for (CpEdge::iterator ite=(*it)->begin();ite!=(*it)->end();ite++)
+        region.insert(&*ite);
+      reduceToRegion(F,region,false);
```

Comment:
Good question. I debugged it for a while and this is the final code I came up with. First of all, the original code of directly deleting basic block crashes. One of the reasons is bad PHI node in successors. To resolve the imprecise implementation of deleteing, I changed it to a existing util function (DeleteDeadBlock) in a commit. This solved the phi node issue, but still crashes in some files. I had a few speculations on the causes. Overall, I think it is better to leave the optimization to an additional pass.